ciao-controller/internal/datastore/datastore.go

```diff
@@ -772,6 +795,24 @@ func (ds *Datastore) AllocateTenantIP(tenantID string) (net.IP, error) {
 
 	ds.tenantsLock.Unlock()
 
+	// must be done with lock unlocked?.
+	mgr := ds.tenants[tenantID].CNCIctrl
+
+	if addSubnet && mgr != nil {
+		err := mgr.AddSubnet(int(subnetInt))
+		if err != nil {
+			glog.Warningf("Error adding subnet (%v)\n", err)
+		}
+	}
+
+	// wait for cnci to become ready before moving on.
+	if mgr != nil {
+		err := mgr.WaitForActive(int(subnetInt))
+		if err != nil {
```

Comment:
Or, after talking to @markdryan if you're just wanting to block then AddSubnet() can just return a channel, that might already be closed and <- on that. You can then use the same channel for all requests into AddSubnet()